core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -124,13 +127,19 @@ private[spark] class LiveListenerBus(conf: SparkConf) {
   }
 
   /** Post an event to all queues. */
-  def post(event: SparkListenerEvent): Unit = {
-    if (!stopped.get()) {
-      metrics.numEventsPosted.inc()
+  def post(event: SparkListenerEvent): Unit = synchronized {
+    if (stopped.get()) {
+      return
+    }
```

Comment:
What's the difference between this and the original way 
```
if (!stopped.get()) {  
...
}
``` 
?

Reply:
It avoids having to indent the rest of the code.

Reply:
Is `synchronized` needed only due to the modification of `queuedEvents`?

Reply:
No. It's needed because you have to make sure that when the buffered events are posted to the queues in `start()`, this method cannot be called. Otherwise you need to have a contract that this class is not thread-safe until after `start()` is called.

Reply:
Let me see if I can rearrange things to avoid the extra synchronization...